### PR TITLE
Refactor and shorten component creation

### PR DESCRIPTION
**Pull request by alexmaco**
It changes the following:
- groups common items for components initialization in an Environment to reduce some repetition

I followed the checklist:
- [ ] I added unittests
- [x] I ran `make check` without errors
- [x] I tested the overall application
- [ ] I added an appropriate item to the changelog

### CODE REVIEW

**Comment by extrawurst** [comment]
I like the idea. I would like to move forward with this. please rebase again and resolve the merge conflicts!

**Comment by alexmaco** [comment]
@extrawurst perhaps another look at this?

File: src/components/changes.rs [hunk 1]
```diff
@@ -35,28 +35,18 @@ impl ChangesComponent {
 	//TODO: fix
 	#[allow(clippy::too_many_arguments)]
```
Comment (extrawurst) [inline src/components/changes.rs:36]: is this fixed by the change? can it be removed?

File: src/components/commit.rs [hunk 2]
```diff
@@ -65,30 +65,25 @@ const FIRST_LINE_LIMIT: usize = 50;
...
 impl CommitComponent {
 	///
-	pub fn new(
-		repo: RepoPathRef,
-		queue: Queue,
-		theme: SharedTheme,
-		key_config: SharedKeyConfig,
-		options: SharedOptions,
-	) -> Self {
+	pub fn new(env: &crate::app::Environment) -> Self {
```
Comment (extrawurst) [inline src/components/commit.rs:68]: so far the other places import this instead of fully-qualifying. lets stay consistent

File: src/app.rs [hunk 3]
```diff
@@ -113,14 +113,41 @@ pub struct App {
...
+	pub key_config: SharedKeyConfig,
+	pub repo: RepoPathRef,
+	pub options: SharedOptions,
+	pub sender_git: Sender<AsyncGitNotification>,
+	pub sender_app: Sender<AsyncAppNotification>,
+}
+
+/// The need to construct a "whatever" environment only arises in testing right now
+#[cfg(test)]
+impl Default for Environment {
```
Comment (extrawurst) [inline src/app.rs:128]: does it need to be the `Default` trait though? lets make it explicit and just make it a static fn `new_test_env`

File: src/app.rs [hunk 4]
```diff
@@ -113,14 +113,41 @@ pub struct App {
...
+	fn default() -> Self {
+		use crossbeam_channel::unbounded;
+		Self {
+			queue: Queue::new(),
+			theme: Default::default(),
+			key_config: Default::default(),
+			repo: RefCell::new(RepoPath::Path(Default::default())),
+			options: Default::default(),
```
Comment (extrawurst) [inline src/app.rs:136]: same for `Options::default`

File: src/options.rs [hunk 5]
```diff
@@ -32,6 +32,17 @@ pub struct Options {
 	data: OptionsData,
 }
 
+#[cfg(test)]
+impl Default for Options {
```
Comment (extrawurst) [inline src/options.rs:36]: see above

**Review by extrawurst — CHANGES_REQUESTED** [review]
obviously just nitpick. good work!

**Comment by extrawurst** [comment]
@alexmaco are you still on this?

**Comment by extrawurst** [comment]
made these cleanups myself now and will merge as part of #2043

**Pull request closed without merging**